tensorrt_llm/_torch/pyexecutor/executor_request_queue.py

```
@@ -232,42 +232,123 @@ def _fetch_new_requests_attention_dp(
+    def _schedule_attention_dp_requests(
+            self, scheduled_requests: List[RequestQueueItem],
+            unscheduled_requests: List[RequestQueueItem]
+    ) -> List[RequestQueueItem]:
+        """Schedule attention dp requests."""
+        new_requests_cur_rank = []
+
+        # Schedule the requests with attention dp rank and no relax
+        for req_item in scheduled_requests:
+            target_dp_rank = req_item.request.py_schedule_params.attention_dp_rank
+            assert self.all_ranks_num_active_requests[target_dp_rank] < self.max_num_active_requests, \
+                f"The number of active requests on rank {target_dp_rank} is {self.all_ranks_num_active_requests[target_dp_rank]}, " \
```

Ruff (0.12.2)

Line too long (133 > 120)

(E501)